crates/state/src/wl_state.rs

```diff
@@ -227,14 +230,24 @@ where
         // hashes from the previous block to the general bucket
         self.move_current_replay_protection_entries(batch)?;
 
-        for hash in
-            std::mem::take(&mut self.0.write_log.replay_protection).iter()
-        {
-            self.write_replay_protection_entry(
-                batch,
-                &replay_protection::current_key(hash),
-            )?;
-        }
+        let replay_prot_key = replay_protection::commitment_key();
+        let commitment: Hash = self
+            .read(&replay_prot_key)
+            .expect("Could not read db")
+            .unwrap_or_default();
+        let new_commitment =
+            std::mem::take(&mut self.0.write_log.replay_protection)
+                .iter()
+                .try_fold(commitment, |mut acc, hash| {
+                    self.write_replay_protection_entry(
+                        batch,
+                        &replay_protection::current_key(hash),
+                    )?;
+                    acc = acc + hash;
+                    Ok::<_, Error>(acc)
+                })?;
+        self.write(&replay_prot_key, new_commitment)?;
```

Comment:
I'm more in favor of manually updating a `Sha256` state here with `try_for_each` vs. `try_fold`, then writing the resulting hash after `try_for_each`, returned from `hasher.finalize()`

Reply:
That wouldn't quite work. We would need to finalize each iteration and create a new hasher. It ends up being needlessly noisy, hence the Add impl.

Reply:
hm I see. maybe we can define/reuse an existing merkle tree hasher fn for two hashes?

Reply:
nvm just saw your last commit!